dfastmi/gui.py

```diff
@@ -315,10 +316,10 @@ def updated_branch(ibranch: int) -> None:
         Newly selected branch number.
     """
     reach = dialog["reach"]
-    reach.clear()
-    reach.addItems(rivers.allreaches[ibranch])
+    reach.clear()    
+    reach.addItems([reach.name for reach in rivers.branches[ibranch].reaches])
 
-    dialog["qloc"].setText(rivers.qlocations[ibranch])
+    dialog["qloc"].setText(rivers.branches[ibranch].qlocation)
```

Comment:
we keep getting back on the index, instead of the branch itself.
A specific reason for it or is that because it was implemented that way and is not yet up for change?

Reply:
I agree, but gui.py is not in scope of this issue. I only refactor code to get it working again. Will pickup the good comments later in new issue

Reply:
We need to create a GUI refactoring issue for this

Reply:
Discussed, we will leave these comments for a later refactoring.
When a new issue is created for this, the number will be disclosed here.